UI/ResearchWnd.cpp

```diff
@@ -172,11 +172,22 @@ namespace {
         m_name_text->SetChildClippingMode(ClipToClient);
         top += m_name_text->Height();
 
-        m_progress_bar = new MultiTurnProgressBar(tech ? tech->ResearchTime(m_empire_id) : 1,
-                                                  turns_completed, tech ? tech->ResearchCost(m_empire_id) : 0.0, turn_spending,
-                                                  GG::LightColor(ClientUI::TechWndProgressBarBackgroundColor()),
-                                                  ClientUI::TechWndProgressBarColor(),
-                                                  m_in_progress ? ClientUI::ResearchableTechFillColor() : GG::LightColor(ClientUI::ResearchableTechFillColor()) );
+        int total_time = 0;
+        float next_progress = 0.0f;
+        if (tech) {
+            total_time = tech->ResearchTime(m_empire_id);
+            float total_cost = tech->ResearchCost(m_empire_id);
+            next_progress = turn_spending / std::max(1.0f, total_cost);
+        }
+        m_progress_bar = new MultiTurnProgressBar(total_time,
+            total_time > 0 ? turns_completed / total_time : 0.0f,
```

Comment:
I you do most of the calculation outside the constructor parameter list you should move all of it.